### PR TITLE
[Button] Add support of button types, added textOnly rebranded color

### PR DESCRIPTION
**Pull request by levadadenys**
* add support of button types to Button.jsx
* add support of rebranded textButton
* needed for https://github.com/code-dot-org/code-dot-org/pull/52038 and future work

Visuals:
Знімок екрана 2023-05-26 о 18 22 01

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/Button.jsx [hunk 1]
```diff
@@ -44,6 +44,7 @@ const ButtonHeight = {
 
 class Button extends React.Component {
   static propTypes = {
+    type: PropTypes.oneOf(['button', 'submit', 'reset']),
```
Comment (rshipp) [inline apps/src/templates/Button.jsx:47]: 🤩 thank you for adding this!!

**Review by rshipp — APPROVED** [review]

File: apps/src/templates/Button.jsx [hunk 2]
```diff
@@ -154,7 +156,9 @@ class Button extends React.Component {
         this.props.className,
         moduleStyles.main,
         moduleStyles.textButton,
-        'button-active-no-border'
+        'button-active-no-border',
+        color === ButtonColor.brandSecondaryDefault &&
+          moduleStyles.rebrendedTextButton
```
Comment (bencodeorg) [inline apps/src/templates/Button.jsx]: typo: rebranded?

**Review by bencodeorg — APPROVED** [review]
LGTM!

File: apps/src/templates/Button.jsx (hunk 2 above)
Comment (levadadenys) [reply apps/src/templates/Button.jsx]: Thanks, will fix!

**Pull request merged**